File: ios/TingShu/Sources/Models/Enums.swift
// Enums mirror `server/app/core/enums.py`. Wire format is the lowercase
// string (`"male"`, `"teen"`, …) so JSON round-trips stay stable.
import Foundation

enum Gender: String, Codable, CaseIterable, Sendable {
    case male
    case female
    case neutral
}

enum Age: String, Codable, CaseIterable, Sendable {
    case child
    case teen
    case youth
    case adult
    case elder
}

enum Personality: String, Codable, CaseIterable, Sendable {
    case calm, gentle, cheerful, serious, cold, fierce
    case determined, timid, playful, mature, naive, wise
    case arrogant, kind, cunning, brave, melancholy, passionate
}

enum Tone: String, Codable, CaseIterable, Sendable {
    case neutral
    case happy
    case sad
    case angry
    case fearful
    case surprised
    case gentle
    case serious
    case playful
    case whisper
}

enum BookStatus: String, Codable, Sendable {
    case uploading
    case processing
    case ready
    case failed
    // Client-only state for books where the server says `ready` but we
    // haven't finished pulling down `meta.json` + chapter texts yet.
    case downloading
}

File: server/app/core/enums.py
"""Predefined enumerations shared between book metadata, voice library and TTS API.

See docs/technical-plan.md §一 for the canonical spec.
"""
from __future__ import annotations

from enum import Enum


class Gender(str, Enum):
    MALE = "male"
    FEMALE = "female"
    NEUTRAL = "neutral"


class Age(str, Enum):
    CHILD = "child"
    TEEN = "teen"
    YOUTH = "youth"
    ADULT = "adult"
    ELDER = "elder"


class Personality(str, Enum):
    CALM = "calm"
    GENTLE = "gentle"
    CHEERFUL = "cheerful"
    SERIOUS = "serious"
    COLD = "cold"
    FIERCE = "fierce"
    DETERMINED = "determined"
    TIMID = "timid"
    PLAYFUL = "playful"
    MATURE = "mature"
    NAIVE = "naive"
    WISE = "wise"
    ARROGANT = "arrogant"
    KIND = "kind"
    CUNNING = "cunning"
    BRAVE = "brave"
    MELANCHOLY = "melancholy"
    PASSIONATE = "passionate"


class Tone(str, Enum):
    NEUTRAL = "neutral"
    HAPPY = "happy"
    SAD = "sad"
    ANGRY = "angry"
    FEARFUL = "fearful"
    SURPRISED = "surprised"
    GENTLE = "gentle"
    SERIOUS = "serious"
    PLAYFUL = "playful"
    WHISPER = "whisper"


class BookStatus(str, Enum):
    UPLOADING = "uploading"
    PROCESSING = "processing"
    READY = "ready"
    FAILED = "failed"
